src/tests/class_tests/openms/source/IdentificationData_test.cpp

```diff
@@ -0,0 +1,386 @@
+  peptide.parent_matches[rna_ref];
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerIdentifiedPeptide(peptide));
+}
+END_SECTION
+
+START_SECTION((const IdentifiedPeptides& getIdentifiedOligos() const))
+{
+  TEST_EQUAL(data.getIdentifiedOligos().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((IdentifiedPeptideRef registerIdentifiedOligo(const IdentifiedOligo& oligo)))
+{
+  IdentificationData::IdentifiedOligo oligo(NASequence::fromString(""));
+  // can't register an oligo without a sequence:
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerIdentifiedOligo(oligo));
+  TEST_EQUAL(data.getIdentifiedOligos().empty(), true);
+
+  // oligo without RNA reference:
+  oligo.sequence = NASequence::fromString("ACGU");
+  oligo_ref = data.registerIdentifiedOligo(oligo);
+  TEST_EQUAL(data.getIdentifiedOligos().size(), 1);
+  TEST_EQUAL(*oligo_ref == oligo, true); // "TEST_EQUAL(*oligo_ref, oligo)" doesn't compile
+
+  // oligo with protein reference:
+  oligo.sequence = NASequence::fromString("UGCA");
+  oligo.parent_matches[rna_ref];
+  oligo_ref = data.registerIdentifiedOligo(oligo);
+  TEST_EQUAL(data.getIdentifiedOligos().size(), 2);
+  TEST_EQUAL(*oligo_ref == oligo, true); // "TEST_EQUAL(*oligo_ref, oligo)" doesn't compile
+
+  // registering an oligo with protein reference doesn't work:
+  oligo.parent_matches[protein_ref];
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerIdentifiedOligo(oligo));
+}
+END_SECTION
+
+START_SECTION((const IdentifiedPeptides& getIdentifiedCompounds() const))
+{
+  TEST_EQUAL(data.getIdentifiedCompounds().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((IdentifiedPeptideRef registerIdentifiedCompound(const IdentifiedCompound& compound)))
+{
+  IdentificationData::IdentifiedCompound compound("");
+  // can't register a compound without identifier:
+  TEST_EXCEPTION(Exception::IllegalArgument,
+                 data.registerIdentifiedCompound(compound));
+  TEST_EQUAL(data.getIdentifiedCompounds().empty(), true);
+
+  compound = IdentificationData::IdentifiedCompound("compound_1",
+                                                    EmpiricalFormula("C2H5OH"),
+                                                    "ethanol");
+  compound_ref = data.registerIdentifiedCompound(compound);
+  TEST_EQUAL(data.getIdentifiedCompounds().size(), 1);
+  TEST_EQUAL(*compound_ref == compound, true); // "TEST_EQUAL(*compound_ref, compound)" doesn't compile
+}
+END_SECTION
+
+START_SECTION((const MoleculeQueryMatches& getMoleculeQueryMatches() const))
+{
+  TEST_EQUAL(data.getMoleculeQueryMatches().empty(), true);
+  // tested further below
+}
+END_SECTION
+
+START_SECTION((QueryMatchRef registerMoleculeQueryMatch(const MoleculeQueryMatch& match)))
+{
+  // match with a peptide:
+  IdentificationData::MoleculeQueryMatch match(peptide_ref, query_ref, 3);
```

Comment:
for readability and testability, can you please make the test self-contained? This now references a peptide_ref that from a different test which is somewhat confusing, it would be better for documentation purposes and separability of the tests to generate a new peptide_ref and query_ref to see what is going on here.

Reply:
Unfortunately this is not so easy. Because of all the cross-references between data, making tests self-contained would lead to a lot of code duplication. So I think it's better to keep the tests incremental (same as a lot of other OpenMS tests).